File: src/clase-05/practica/python/activity3_decorator_starter/decorators.py
from notifier import Notifier

class NotifierDecorator(Notifier):
    def __init__(self, notifier: Notifier):
        self._notifier = notifier

    def send(self, message: str) -> str:
        return self._notifier.send(message)

class EmailDecorator(NotifierDecorator):
    def send(self, message: str) -> str:
        base = super().send(message)
        return f"{base} | canal=Email"

class SMSDecorator(NotifierDecorator):
    def send(self, message: str) -> str:
        base = super().send(message)
        return f"{base} | canal=SMS"

class SlackDecorator(NotifierDecorator):
    def send(self, message: str) -> str:
        base = super().send(message)
        return f"{base} | canal=Slack"

File: src/clase-05/practica/typescript/activity3_decorator_starter/decorators.ts
import { Notifier } from "./notifier";

export class NotifierDecorator extends Notifier {
  constructor(protected notifier: Notifier) {
    super();
  }
  send(message: string): string {
    return this.notifier.send(message);
  }
}

export class EmailDecorator extends NotifierDecorator {
  send(message: string): string {
    return `${super.send(message)} | canal=Email`;
  }
}

export class SMSDecorator extends NotifierDecorator {
  send(message: string): string {
    return `${super.send(message)} | canal=SMS`;
  }
}

export class SlackDecorator extends NotifierDecorator {
  send(message: string): string {
    return `${super.send(message)} | canal=Slack`;
  }
}
